# World Happyness Report - The Course No1 Project


In [1]:
# Import usefull libraries for the project
import matplotlib.pyplot as plt
%matplotlib inline

## A. Importing, cleaning and numerical summaries

### Load and analyze the data

First, we build a pandas dataframe object by loading the data.csv file.

In [2]:
import os
import pandas as pd

datafile_path = os.path.join('data','data.csv')

df = pd.read_csv(datafile_path)


Now it's time to grab some informations from this freshly loaded dataframe.

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 153 entries, 0 to 152
Data columns (total 12 columns):
Country             153 non-null object
Happiness Rank      153 non-null int64
Happiness Score     153 non-null float64
Economy             153 non-null float64
Family              153 non-null float64
Health              153 non-null float64
Freedom             153 non-null float64
Generosity          153 non-null float64
Corruption          153 non-null float64
Dystopia            153 non-null float64
Job Satisfaction    151 non-null float64
Region              153 non-null object
dtypes: float64(9), int64(1), object(2)
memory usage: 14.4+ KB


This dataframe is made of a total of 12 columns and 153 lines.

The df.info() result confirm that there is no missing data in this dataste as all the columns contains exactly the same 'non-null' values as the total of lines in the dataset.
Columns types are floats (values of the observations), integer (for the country rank) and 2 generic types (Country names and Regions)


### Reindex the dataset

Displaying a few lines of the dataframe using a `df.head(5)` function call, we can see that index could be rearranged using the Country Name.
This can be easily done using the following code.

_Do not forget the `inplace=True` to update our dataframe_



In [4]:
df.set_index('Country',inplace=True)
df.head(3)

Happiness Rank  Happiness Score   Economy    Family    Health  \
Country                                                                   
Malta                 27            6.527  1.343280  1.488412  0.821944   
Zimbabwe             138            3.875  0.375847  1.083096  0.196764   
Cyprus                65            5.621  1.355938  1.131363  0.844715   

           Freedom  Generosity  Corruption  Dystopia  Job Satisfaction  \
Country                                                                  
Malta     0.588767    0.574731    0.153066  1.556863              85.2   
Zimbabwe  0.336384    0.189143    0.095375  1.597970              56.3   
Cyprus    0.355112    0.271254    0.041238  1.621249              88.7   

                  Region  
Country                   
Malta     Western Europe  
Zimbabwe          Africa  
Cyprus    Eastern Europe

At that point, our dataframe contains 11 column as the `'Country'` one became its index.

In [7]:
df.columns

Index(['Happiness Rank', 'Happiness Score', 'Economy', 'Family', 'Health',
       'Freedom', 'Generosity', 'Corruption', 'Dystopia', 'Job Satisfaction',
       'Region'],
      dtype='object')

### Grab some stats

Basic statistics values of the data can be obtain using the `describe()` function.

We can quickly restrict the statistcs displayed to the mean, the min and the max values of each numerical columns using the `.loc()`function.

In [19]:
df.describe().loc[['mean','min','max']]

Happiness Rank  Happiness Score   Economy    Family    Health   Freedom  \
mean       78.169935         5.349281  0.982433  1.186630  0.550117  0.408489   
min         1.000000         2.693000  0.000000  0.000000  0.000000  0.000000   
max       155.000000         7.537000  1.870766  1.610574  0.949492  0.658249   

      Generosity  Corruption  Dystopia  Job Satisfaction  
mean    0.245324    0.123179  1.853072         75.209934  
min     0.000000    0.000000  0.377914         44.400000  
max     0.838075    0.464308  3.117485         95.100000

In [20]:
df.sort_values('Happiness Rank')

Happiness Rank  Happiness Score   Economy    Family  \
Country                                                                         
Central African Republic             155            2.693  0.000000  0.000000   
Somalia                               93            5.151  0.022643  0.721151   
Burundi                              154            2.905  0.091623  0.629794   
Congo (Kinshasa)                     126            4.280  0.092102  1.229023   
Liberia                              148            3.533  0.119042  0.872118   
Niger                                135            4.028  0.161925  0.993025   
Malawi                               136            3.970  0.233442  0.512569   
Mozambique                           113            4.550  0.234306  0.870701   
Guinea                               149            3.507  0.244550  0.791245   
Togo                                 150            3.495  0.305445  0.431883   
Madagascar                           144            3.644  0.305809  0.913020   
Ethiopia                             119            4.460  0.339234  0.864669   
Burkina Faso                         134            4.032  0.350228  1.043280   
Myanmar                              114            4.545  0.367111  1.123236   
Sierra Leone                         106            4.709  0.368421  0.984136   
Haiti                                145            3.603  0.368610  0.640450   
Rwanda                               151            3.471  0.368746  0.945707   
Zimbabwe                             138            3.875  0.375847  1.083096   
Uganda                               133            4.081  0.381431  1.129828   
South Sudan                          147            3.591  0.397249  0.601323   
Afghanistan                          141            3.794  0.401477  0.581543   
Benin                                143            3.657  0.431085  0.435300   
Chad                                 137            3.936  0.438013  0.953856   
Mali                                 127            4.190  0.476180  1.281473   
Senegal                              115            4.535  0.479309  1.179692   
Nepal                                 99            4.962  0.479820  1.179283   
Tanzania                             153            3.349  0.511136  1.041990   
Lesotho                              139            3.808  0.521021  1.190095   
Tajikistan                            96            5.041  0.524714  1.271463   
Kenya                                112            4.553  0.560479  1.067951   
...                                  ...              ...       ...       ...   
Israel                                11            7.213  1.375382  1.376290   
Spain                                 34            6.403  1.384398  1.532091   
Italy                                 48            5.964  1.395067  1.444923   
South Korea                           55            5.838  1.401678  1.128274   
New Zealand                            8            7.314  1.405706  1.548195   
Japan                                 51            5.920  1.416915  1.436338   
France                                31            6.442  1.430923  1.387777   
United Kingdom                        19            6.714  1.441634  1.496460   
Finland                                5            7.469  1.443572  1.540247   
Belgium                               17            6.891  1.463781  1.462313   
Canada                                 7            7.316  1.479204  1.481349   
Iceland                                3            7.504  1.480633  1.610574   
Denmark                                2            7.522  1.482383  1.551122   
Australia                             10            7.284  1.484415  1.510042   
Austria                               13            7.006  1.487097  1.459945   
Germany                               16            6.951  1.487923  1.472520   
Bahrain                               41            6.087